File: week02/Valid Anagram.java
// TC: O(N)
// MC: O(1)

class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length() != t.length()) return false;
        
        int[] counter = new int[26];
        for(int i = 0; i<t.length(); ++i) {
            counter[s.charAt(i)-'a']++;
            counter[t.charAt(i)-'a']--;
        }
        
        for(int c: counter) {
            if(c != 0) return false;
        }
        return true;
    }
}

File: week02/validAnagram.cpp
class Solution
{
public:
    int MOD = 1000000007;
    int offset = 233;

    long long getHash(string &str)
    {
        long long hashValue = 1;
        for (int i = 0; i < str.size(); i++)
        {
            hashValue = hashValue * (str[i] - 'a' + offset) % MOD;
        }
        return hashValue;
    }

    bool isAnagram(string s, string t)
    {
        return (getHash(s) == getHash(t));
    }
};